src/Table/Synapse/Assert.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\TableBackendUtils\Table\Synapse;
+
+use LogicException;
+
+final class Assert
+{
+    public static function assertTableIndex(string $indexName): void
+    {
+        if (!in_array($indexName, [
+            TableIndexDefinition::TABLE_INDEX_TYPE_CCI,
+            TableIndexDefinition::TABLE_INDEX_TYPE_HEAP,
+            TableIndexDefinition::TABLE_INDEX_TYPE_CI,
+        ], true)) {
+            throw new LogicException(sprintf(
+                'Unknown table index type: "%s" specified.',
+                $indexName
+            ));
+        }
+    }
+
+    /**
+     * @param String[] $indexedColumnsNames
+     */
+    public static function assertValidClusteredIndex(string $indexName, array $indexedColumnsNames): void
+    {
+    }
+
+    /**
+     * @param string $tableDistributionName
+     * @param string[] $hashDistributionColumnsNames
+     */
+    public static function assertValidHashDistribution(
+        string $tableDistributionName,
+        array $hashDistributionColumnsNames
+    ): void {
+        if ($tableDistributionName === TableDistributionDefinition::TABLE_DISTRIBUTION_HASH
+            && count($hashDistributionColumnsNames) !== 1
+        ) {
+            throw new LogicException('HASH table distribution must have one distribution key specified.');
+        }
+    }
+
+    public static function assertTableDistribution(string $tableDistributionName): void
+    {
+        if (!in_array($tableDistributionName, [
+            TableDistributionDefinition::TABLE_DISTRIBUTION_HASH,
```

Comment:
nemohlo by vzniknout něco jako TABLE_DISTRIBUTIONS_AVAILABLE (jako pole sdružující tyto hodnoty) nebo tak něco, ať to můžu pak použít v RO a může to být i tady.

Reply:
přidal sem do obou `AVAILABLE_TABLE_*` ať to nemá stejný prefix jak ostatní konstanty špatně by to validoval stan.